pulsar-common/src/main/java/org/apache/pulsar/common/util/SimpleTextOutputStream.java

```diff
@@ -43,19 +44,16 @@ public SimpleTextOutputStream write(byte[] a, int offset, int len) {
     }
 
     public SimpleTextOutputStream write(char c) {
-        buffer.writeByte((byte) c);
+        write(String.valueOf(c));
```

Comment:
I wouldn't do that, since we're back to the same bug if some future developer decides to use it wrong.
I prefer to abolish that method altogether and refactor it to use String instead.